components/backup-plugins/internal/plugins/nats-channels.go

```diff
@@ -0,0 +1,99 @@
+	messagingv1alpha1Client "github.com/knative/eventing/pkg/client/clientset/versioned/typed/messaging/v1alpha1"
+	apierrors "k8s.io/apimachinery/pkg/api/errors"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/client-go/kubernetes"
+	"k8s.io/client-go/rest"
+)
+
+const (
+	channelNameSuffix = "-kn-channel"
+	channelKind       = "NatssChannel"
+)
+
+// SetOwnerReference is a plugin for velero to set new value of UID in metadata.ownerReference field
+// based on new values of other restored objects
+type SetNatsChannelOwnerReference struct {
+	Log logrus.FieldLogger
+}
+
+// AppliesTo return list of resource kinds which should be handled by this plugin
+func (p *SetNatsChannelOwnerReference) AppliesTo() (velero.ResourceSelector, error) {
+	return velero.ResourceSelector{
+		IncludedResources: []string{"service"},
+		LabelSelector:     "messaging.knative.dev/role=natss-channel",
+	}, nil
+}
+
+// Execute contains main logic for plugin
+// nolint
+func (p *SetNatsChannelOwnerReference) Execute(input *velero.RestoreItemActionExecuteInput) (
+	*velero.RestoreItemActionExecuteOutput, error) {
+	metadata, err := meta.Accessor(input.Item)
+	if err != nil {
+		return nil, err
+	}
+	config, err := rest.InClusterConfig()
+	if err != nil {
+		p.Log.Infof("ERROR: getting cluster config: %v", err)
+		return nil, err
+	}
+	p.Log.Infof("Setting owner reference for %s %s in namespace %s", input.Item.GetObjectKind(),
+		metadata.GetName(), metadata.GetNamespace())
+
+	evClient, err := evclientset.NewForConfig(config)
+	evClient.MessagingV1alpha1()
+	var messagingChannel messagingv1alpha1Client.MessagingV1alpha1Interface
+	messagingChannel = evClient.MessagingV1alpha1()
+	channelName := metadata.GetName() + channelNameSuffix
+	p.Log.Infof("NATS Channel Owner: %s", channelName)
+	natsChannel, err := messagingChannel.Channels(metadata.GetNamespace()).Get(channelName, metav1.GetOptions{})
+	switch {
+	case err == nil:
+	case apierrors.IsNotFound(err):
+		// there's no NATS Channel with such name
+		p.Log.Errorf("Couldn't get NATS Channel %s: %v", metadata.GetName(), err)
+		return velero.NewRestoreItemActionExecuteOutput(input.Item), nil
+	default:
+		return velero.NewRestoreItemActionExecuteOutput(input.Item), err
+	}
+	var natsChannelKind = v1alpha1.SchemeGroupVersion.WithKind(channelKind)
+
+	ownerReferences := []metav1.OwnerReference{
+		*metav1.NewControllerRef(natsChannel, natsChannelKind),
+	}
+	metadata.SetOwnerReferences(ownerReferences)
+
+	return velero.NewRestoreItemActionExecuteOutput(input.Item), nil
+}
+
+func (p *SetNatsChannelOwnerReference) inClusterClient() (*kubernetes.Clientset, error) {
+	config, err := p.inClusterConfig()
+	if err != nil {
+		return nil, err
+	}
+	clientset, err := kubernetes.NewForConfig(config)
+	if err != nil {
+		p.Log.Errorf("Error in creating the clientset: %v", err.Error())
+		return nil, err
+	}
+
+	return clientset, nil
+}
+
+func (p *SetNatsChannelOwnerReference) inClusterConfig() (*rest.Config, error) {
```

Comment:
What is the purpose of this function? Why not directly use `rest.InClusterConfig()` in line `79`?

Reply:
Just separation of concerns, it was inspired by https://github.com/kyma-project/kyma/blob/master/components/backup-plugins/internal/plugins/owner_reference.go#L70

Reply:
This function just wraps another function with the exact same signature.
Calling `rest.InClusterConfig()` is already separating the concern of getting the in-cluster config to the `rest` package.

I am sorry, but I do not see how `inClusterConfig()` has a responsability.

Reply:
OK, thanks!
Please recheck now.